src/init/caches.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <init/caches.h>
+
+#include <txdb.h>
+#include <util/system.h>
+#include <validation.h>
+
+void CalculateCacheSizes(const ArgsManager& args, CacheSizes& cache_sizes, size_t n_indexes)
```

Comment:
1f966b8d9f

One naive question here might be: why are we using a mutable arg for `cache_sizes` vs. just returning a constructed `CacheSizes` instance to make the function pure? Answer seems to be that this is used within a unittest context to modify an existing `m_cache_sizes` instance.

Reply:
Fixed as of 6736e24a490d756e0e1e70b9e212775cf305ebe6